pages/studentPortal/web/React/assignments/tailwindAssignment.tsx

```diff
@@ -0,0 +1,184 @@
+          if (data.user_assignment_submissions.length > 0) {
+            dispatch(setUserAssignments(data.user_assignment_submissions));
+            deployCurrentStage(data.user_assignment_submissions[0]);
+          }
+        },
+      }
+    );
+
+  const completedStage: AssignmentComponentData[] = [
+    {
+      component: "loom-video",
+      text: "Your assignment has been reviewed! Watch the video below for feedback:",
+      videoId: userAssignments.length > 0 ? userAssignments[0].review_link : "",
+    },
+  ];
+
+  const handleContinue = () => {
+    router.push("/studentPortal/web/React/components");
+  };
+  return (
+    <>
+      <div className="flex flex-col m-8 space-y-4 sm:px-12">
+        {userAssignmentsLoading ? (
+          <div>Loading...</div>
+        ) : stage === Stage.INCOMPLETE ? (
+          incompleteStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : stage === Stage.SUBMITTED ? (
+          submittedStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : stage === Stage.COMPLETED ? (
+          completedStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : null}
+        <div className="flex my-8 mr-8 sm:justify-end">
+          <Button onClick={handleContinue} label="Continue" />
+        </div>
+      </div>
+    </>
+  );
+};
+export async function getServerSideProps() {
+  const assignmentId = "2cf9156a-4f6f-452d-b09a-2c54f19a7b40";
+
+  const incompleteStage: AssignmentComponentData[] = [
+    {
+      component: "title",
+      text: "Assignment: Tailwind CSS",
+    },
+
+    {
+      component: "prompt",
+      header: "Style a web page using Tailwind CSS",
+      bullets: [
+        "Use the link in the first hint to setup TailwindCSS in your own index.html file for this assignment.",
+        "Your page must have atleast 5 different background colors.",
+        "Add a Navbar at the top-right of the page.",
+        "Add a title, a heading and a paragraph so your page has atleast 3 different text sizes.",
+        "Add an on-hover effect to the title, so it bounces when you hover over it.",
+        "Add a table with 5 columns and 4 rows, below the paragraph.",
+        "Stretch challenge: Find a website whose design you like and rebuild it using TailwindCSS. Start from the top and work your way downwards. Speak to a coach if you get stuck!",
+      ],
+    },
+    {
+      component: "submission",
+      codeSandboxTitle:
+        "Please submit your assignment by pasting a link in the box below.",
+      placeholder: "Assignment link goes here",
+      assignmentId: assignmentId,
+      link: "",
+    },
+    {
+      component: "hint-list",
+      hintRow: [
+        {
+          description: "Setup TailwindCSS using Play CDN, follow step one",
```

Comment:
let's abstract the tailwind setup from this assignment,

you can fork this [template](https://codesandbox.io/s/react-typescript-tailwind-template-6t0ywc) then setup the assignment: